stm/vernac_classifier.ml

```diff
@@ -176,7 +176,7 @@ let classify_vernac e =
     (* These commands alter the parser *)
     | VernacOpenCloseScope _ | VernacDelimiters _ | VernacBindScope _
     | VernacInfix _ | VernacNotation _ | VernacNotationAddFormat _
-    | VernacSyntaxExtension _
+    | VernacSyntaxExtension _ | VernacDeclareCustomEntry _
```

Comment:
Does declaring an entry change the parser in a visible way?
Adding notation to that entry do, of course. But just declaring the entry seems not to require `VtNow`.

Reply:
OK, I moved to `VtLater`. Note that this is relevant for #6247, so I'm changing it there.

Reply:
I made the change but now I have a doubt. Where should I indicate that a `Notation` referring to a custom entry `foo` should be evaluated only after the corresponding `Declare Custom Entry foo` has been evaluated.

Reply:
I guess the most important thing is that the error message is clear, eg "No custom entry named 'foo'".

Reply:
If the command doesn't modify the actual grammar parsed `VtLater` tends to be safe I think.

Reply:
Given that `Notation` is `VtNow` the STM (of today) will evaluate all the commands coming before `Notation` in order to execute `Notation`, hence `Notation` is executed in a state where the entry is declared (assuming there is a command `Declare Custom..` before the `Notation` populating the new entry).

In the model we whish to obtain one day, both `Notation` and `Declare Custom Entry` will be commands that have their execution broken into two pieces: changing the parser + the rest.
In the case of `Declare Custom Entry` the "rest" is likely to be empty, while for `Notation` the rest is going to be about checking the interpretation and add the notation to other data structures like the ones used for printing. Both change some data structures relative to parsing, even if only `Notation` in a visible way.

So in a sense `Declare Custom Entry` is closer to `Notation`, but as of today putting `VtNow` (whilst being safe) is not needed. You can leave it if you prefer, maybe with a comment "not strictly needed since just adding a non terminal does not change the language accepted by the parser".

Reply:
Very clear explanation, thanks. I did not realize that the current implementation forces evaluation of all commands preceding a `Notation`.